src/core/frames/frame_redirector.ts

```diff
@@ -47,14 +47,14 @@ export class FrameRedirector implements LinkInterceptorDelegate, FormInterceptor
 
   private shouldRedirect(element: Element, submitter?: HTMLElement) {
     const frame = this.findFrameElement(element)
-    return frame ? frame != element.closest("turbo-frame") : false
+    return frame ? frame != element.closest(`turbo-frame, [is^="turbo-frame-"]`) : false
```

Comment:
Would a check like `[is~="turbo-frame"]` without the corresponding call to `Turbo.defineCustomFrameElement('tbody')` treat `<tbody is="turbo-frame">` elements the same as `<turbo-frame>`?

Reply:
Sadly not, the custom elements API would just ignore it, so the element wouldn’t actually be a working frame.